File: CalciulatorApp/CalciulatorAppTests/CalculatorTests.m
#import "CalculatorTests.h"
#import "Calculator.h"

@implementation CalculatorTests

- (void)setUp {
    [super setUp];
    sut = [[Calculator alloc] init];
}

- (void)testAddMethod_zeroLengthInput_returnsZero {
    NSInteger expected = 0;
    NSInteger result = [sut add:@""];

    STAssertEquals(expected, result, @"Zero length input should return 0.");
}

- (void)testAddMethod_oneLengthInput_returnsEquivalent {
    NSInteger expected = 3;
    NSInteger result = [sut add:@"3"];

    STAssertEquals(expected, result, @"One length input should return equivalent.");
}

- (void)testAddMethod_twoLengthInput_returnsSum {
    NSInteger expected = 8;
    NSInteger result = [sut add:@"3,5"];

    STAssertEquals(expected, result, @"Two length input should return sum.");
}

- (void)testAddMethod_anyLengthInput_returnsSum {
    NSInteger expected = 0;
    NSInteger howMany = arc4random() % 999;
    NSMutableString *numbersToAdd = [NSMutableString string];

    for (int i = 0; i < howMany; i++) {
        [numbersToAdd appendFormat:[NSString stringWithFormat:@"%d,",i]];
        expected += i;
    }

    NSInteger result = [sut add:numbersToAdd];

    STAssertEquals(expected, result, @"Any length input should return sum.");
}


@end

File: CalculatorApp/CalculatorTests.py
from random import randint
from Calculator import Calculator

__author__ = 'dgadd'

import unittest


class MyTestCase(unittest.TestCase):
    def test_addMethod_zeroLengthInput_returnsZero(self):
        expected = 0
        sut = Calculator()
        result = sut.add("")
        self.assertEqual(expected, result, msg="Zero length input should return 0.")

    def test_addMethod_oneLengthInput_returnsEquivalent(self):
        expected = 8
        sut = Calculator()
        result = sut.add("8")
        self.assertEqual(expected, result, msg="One length input should return equivalent.")

    def test_addMethod_twoLengthInput_returnsSum(self):
        expected = 10
        sut = Calculator()
        result = sut.add("3,7")
        self.assertEqual(expected, result, msg="Two length input should return sum.")

    def test_addMethod_anyLengthInput_returnsSum(self):
        randomNumber = randint(111,999)
        expected = 0
        numbersToAdd = ""
        for i in range(111, randomNumber):
            numbersToAdd += "%s," % (i)
            expected += i

        sut = Calculator()
        result = sut.add(numbersToAdd)
        self.assertEqual(expected, result, msg="Any length input should return sum.")

if __name__ == '__main__':
    unittest.main()
